src/main/java/bowling/Frame.java

```diff
@@ -0,0 +1,119 @@
+package bowling;
+
+import bowling.exception.BowlingException;
+import bowling.exception.FrameOverException;
+import bowling.exception.InvalidPinQuantityException;
+
+public class Frame {
+    public static final int MAX_PINS = 10;
+    private final boolean isFinalFrame;
+    private int pinsStanding = MAX_PINS;
+    private int rollNumber = 0;
+    private boolean isOver = false;
+    private int rollOnePins = 0;
+    private int rollTwoPins = 0;
+    private FrameType frameType = FrameType.NORMAL;
+    private int score = 0;
+
+    public Frame(boolean isFinalFrame) {
+        this.isFinalFrame = isFinalFrame;
+    }
+
+    public Frame() {
+        this(false);
+    }
+
+    public void roll(int numberOfPinsKnockedDown) throws BowlingException {
```

Comment:
Some developers are not a fan of a.) using checked exceptions, and b.) creating their own exceptions. I think it's good to re-use the existing Exception types provided by Java (e.g if you try to roll and the game is already over, then throw an InvalidOperationException)

Reply:
+1 on checked exceptions
own exceptions makes sense as it can carry extra context, but I agree the standard ones should be prefered.